buffer/src/main/java/io/netty/buffer/ReferenceCountHolder.java

```diff
@@ -0,0 +1,230 @@
+package io.netty.buffer;
+
+import io.netty.util.IllegalReferenceCountException;
+import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.ReferenceCountUpdater;
+
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+import static io.netty.util.internal.ObjectUtil.checkPositive;
+
+abstract class ReferenceCountHolder {
+    // TODO: inline
+    private static final ReferenceCountUpdater.Configuration<ReferenceCountHolder> config = ReferenceCountUpdater.Configuration.of(
+            ReferenceCountHolder.class,
+            "refCnt",
+            MethodHandles::lookup,
+            AtomicIntegerFieldUpdater::newUpdater
+    );
+    private static final AtomicIntegerFieldUpdater<ReferenceCountHolder> UPDATER = config.updater(); // todo: fixed initializer for NI
+    private static final long OFFSET = config.fieldOffset(); // todo: recompute offset for NI
+    private static final ReferenceCountUpdater.Configuration.UpdaterType TYPE = config.updaterType();
```

Comment:
this won't work well sadly for hotspot; it would be better to use an int or anything different

Reply:
Yep I thought so, I think we need to get rid of the config class anyway